library/src/main/java/com/datatorrent/lib/stream/StreamMerger.java

```diff
@@ -41,35 +47,57 @@
 @Stateless
 public class StreamMerger<K> extends BaseOperator
 {
-  /**
-   * Data input port 1.
-   */
-  public final transient DefaultInputPort<K> data1 = new DefaultInputPort<K>()
+  private int portCount = 0;
+  private final transient ArrayList<DefaultInputPort<K>> ports = new ArrayList<>(portCount);
```

Comment:
A great question, was in the process of doing so. Short answer is it fails wonderfully:
```
java.lang.IllegalArgumentException: Port is not associated to any operator in the DAG: com.datatorrent.lib.stream.StreamMerger$1@3b084709
	at com.datatorrent.stram.plan.logical.LogicalPlan.assertGetPortMeta(LogicalPlan.java:1501)
	at com.datatorrent.stram.plan.logical.LogicalPlan.access$1000(LogicalPlan.java:119)
```

The relevant question is thus is there any way to define ports in this way, e.g. an array of ports, such that the DAG is aware of the ports?